recipes-devtools/openocd/openocd_%.bbappend

```diff
@@ -0,0 +1,8 @@
+SRC_URI = " \
```

Comment:
Make SRC_URI also  with overrides so we can ensure that meta-riscv can be used in multi-BSP scenarios

Reply:
I can't get the override to work as we it to apply to openocd-native. Any ideas?

Reply:
lets define a separate recipe `openocd_riscv.bb` and then in layer.conf set 
```
PREFERRED_VERSION_openocd-native = "riscv"
PREFERRED_VERSION_openocd = "riscv"
```

Reply:
Works for me. Now the problem is that we don't have a conf file for qemuriscv64 in meta-riscv

Reply:
yes, qemuriscv64 is assumed to be working without meta-riscv atm so adding this in meta-riscv makes it depend on this layer, layer can only help the machines, which it maintains, so this could ideally be done for oe-core but it wont be accepted. So questions I have is that

1. Can we use oe-core version for native ? if not then can we generate patches that can be applied to native version in core ?

Reply:
The native version is the important part. In this case I want to use OpenOCD on my native machine to debug a board.

I think the diff is fairly large, it won't be just a few simple patches.

It probably doesn't matter though. Users can use the `U540` machine to target 64-bit boards and `qemuriscv32` to target 32-bit ones.

Reply:
I just pushed an update that works for me.

It doesn't set anything for `qemuriscv64` machine, but I think that's ok.

Reply:
Sounds good